### PR TITLE
Add StoreMessage to DevTools.

### PR DESCRIPTION
**Pull request by SHeimlich**
- Add a StoreMessage subclass of DevToolsMessage.
- Bring StoreMessage in line with the DevTools Communication Protocol.
- Breakdown different Operations types to send better JSON to the client.

cc: @piotrswigon @sjmiles

### CODE REVIEW

File: java/arcs/android/devtools/StoreMessage.kt [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+import arcs.core.data.RawEntity
+import arcs.core.data.util.ReferencablePrimitive
+import arcs.core.storage.ProxyMessage
+import arcs.core.util.JsonValue
+
+/**
+ * An implementation of [DevToolsMessage] to inform DevTools that a [Store] received a
+ * [CrdtOperation].
+ */
+class StoreMessage(
```
Comment (piotrswigon) [inline java/arcs/android/devtools/StoreMessage.kt:19]: Nice job!

Something that I would like you to be aware of / keep an eye on longer term:

In the medium term as we resurrect Web-bits, we should aim to have the protocol defined in an implementation independent way and be able to use the same Developer Tools in both Android and Web embeddings of Arcs.

So while it's ok to have the messages defined in code for the short term (including all of Q4 I think), we should avoid putting in things that are implementation specific already.

File: java/arcs/android/devtools/StoreMessage.kt [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    private val actualMessage: ProxyMessage.Operations<CrdtData, CrdtOperation, Any?>
+) : DevToolsMessage {
+    override val kind: String = STORE_MESSAGE
+    override val message: JsonValue<*>
+        get() = JsonValue.JsonObject(mapOf<String, JsonValue<*>>(
+            "id" to JsonValue.JsonNumber(actualMessage.id?.toDouble() ?: 0.0),
+            "operations" to JsonValue.JsonArray(getMessageAsList())
+        ))
```
Comment (jasonwyatt) [inline java/arcs/android/devtools/StoreMessage.kt:27]:
```suggestion
        get() = JsonValue.JsonObject(
            mapOf<String, JsonValue<*>>(
                "id" to JsonValue.JsonNumber(actualMessage.id?.toDouble() ?: 0.0),
                "operations" to JsonValue.JsonArray(getMessageAsList())
            )
        )
```

Comment (jasonwyatt) [reply java/arcs/android/devtools/StoreMessage.kt:27]: Also looks like there are several places in this file where this type of nit needs to be picked (see: getMessageAsList, for example)

**Review by jasonwyatt — APPROVED** [review]

File: java/arcs/android/devtools/StoreMessage.kt (hunk 2 above)
Comment (SHeimlich) [reply java/arcs/android/devtools/StoreMessage.kt]: Fixed!

**Comment by SHeimlich** [comment]
Copybara manual works in cl/331066622

**Pull request merged**